google/cloud/bigquery/job/query.py

```diff
@@ -930,6 +930,15 @@ def query(self):
             self._properties, ["configuration", "query", "query"]
         )
 
+    @property
+    def query_id(self) -> Optional[str]:
+        """[Preview] ID of a completed query.
+
+        This ID is auto-generated and not guaranteed to be populated.
+        """
+        query_results = self._query_results
+        return query_results.query_id if query_results is not None else None
```

Comment:
Is attribute `query_results.query_id` guaranteed to exist? I see that `query_results` is of class `_QueryResults`, but I don't see this attribute in it.

Reply:
I just added it in this PR. `google/cloud/bigquery/query.py`, below.

Reply:
Thank you!